File: src/Materials/Cell.ts
import { TerrainType } from './TerrainType';

export interface ICell {
  index: number;
  positionY: number;
  positionX: number;
  left: Cell | null;
  right: Cell | null;
  up: Cell | null;
  down: Cell | null;
  coins?: number;
  terrainType: TerrainType;
}

export class Cell {
  index: number;
  positionY: number;
  positionX: number;
  left: Cell | null;
  right: Cell | null;
  up: Cell | null;
  down: Cell | null;
  coins?: number;
  terrainType: TerrainType;

  constructor(props: ICell) {
    const { index, positionX, positionY, left, right, up, down, coins, terrainType } = props;
    this.index = index;
    this.positionX = positionX;
    this.positionY = positionY;
    this.left = left;
    this.right = right;
    this.up = up;
    this.down = down;
    this.coins = coins;
    this.terrainType = terrainType;

  }

}

File: server/cell.py
# class Cell:

#     up = None
#     down = None
#     right = None
#     left = None

#     def __init__(self, data, indexX, indexY):
#         self.data: int = data
#         self.indexX: int = indexX
#         self.indexY: int = indexY
        


class Cell:

    def __init__(self, data: int, terrain: str, positionX: int, positionY: int):
        self.data: int = data
        self.terrain: str = terrain
        self.positionX: int = positionX
        self.positionY: int = positionY    
